lib/util/index.coffee

```diff
@@ -39,6 +40,26 @@ exports.treatAsMissingDevice = (uuidOrId) ->
 		replacementErr.stack = err.stack
 		throw replacementErr
 
+exports.osVersionCompare = (versionA, versionB) ->
+	semverResult = semver.rcompare(versionA, versionB)
+	if semverResult != 0
+		return semverResult
+
+	revA = getRev(versionA)
+	revB = getRev(versionB)
+
+	return revB - revA
```

Comment:
NIT: this coerces `null` to `0` which works but is slightly ugly, I'd just return `0` from `getRev`